### PR TITLE
added a SampleHumanServicesTest class

### PR DESCRIPTION
**Pull request by josephbate**
# Pull Requests Requirements

- [ ] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3 Styleguide and design
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing Guidelines of this project.

## Summary
added a SampleHumanServicesTest class
## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java:1]: The test class would benefit from using Java constants for test data values. Here's how you could refactor the repeated test data:

```java
private static final String PATIENT_FIRSTNAME = "John";
private static final String PATIENT_LASTNAME = "Doe";
private static final String PROVIDER_FIRSTNAME = "Jane";
private static final String PROVIDER_LASTNAME = "Loo";
private static final String PATIENT_BIRTHDATE = "03/06/1993";
private static final String SAMPLE_ACCESSION_NUMBER = "12345";
private static final String PATIENT_GENDER = "M";
private static final String SAMPLE_RECEIVED_TIMESTAMP = "012/06/2024";
private static final String PROVIDER_TYPE = "Physic";
private static final String SAMPLE_ENTERED_DATE = "2024-06-03";
```
Constants in Java, declared with `static final` and named in UPPERCASE_SNAKE_CASE, improve code maintainability, reduce typos, and follow the "Single Source of Truth" principle. They simplify mass updates, document magic values, and enable reuse across multiple test methods.

**Review by mherman22 — CHANGES_REQUESTED** [review]
LGTM if you fix the comment below

Thanks @josephbate

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java
Comment (josephbate) [reply src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java:1]: thank you @mherman22 for this insight let make these changes and comit

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(SAMPLE_RECEIVED_TIMESTAMP);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(SAMPLE_ENTERED_DATE);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(SAMPLE_RECEIVED_TIMESTAMP).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(PROVIDER_LASTNAME);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(PATIENT_BIRTHDATE);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(PATIENT_BIRTHDATE).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(lastname2);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(birthdate);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(birthdate).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(receivedTimestamp);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(entereddate);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(receivedTimestamp).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(PROVIDER_LASTNAME);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(PATIENT_BIRTHDATE);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(PATIENT_BIRTHDATE).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(SAMPLE_RECEIVED_TIMESTAMP);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(SAMPLE_ENTERED_DATE);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(SAMPLE_RECEIVED_TIMESTAMP).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(PROVIDER_LASTNAME);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(PATIENT_BIRTHDATE);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(PATIENT_BIRTHDATE).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(SAMPLE_RECEIVED_TIMESTAMP);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(SAMPLE_ENTERED_DATE);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(SAMPLE_RECEIVED_TIMESTAMP).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(PROVIDER_LASTNAME);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(PATIENT_BIRTHDATE);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(PATIENT_BIRTHDATE).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(SAMPLE_RECEIVED_TIMESTAMP);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(SAMPLE_ENTERED_DATE);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(SAMPLE_RECEIVED_TIMESTAMP).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,440 @@
...
+        person2.setLastName(PROVIDER_LASTNAME);
+        personService.save(person2);
+
+        DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
+        Date date = dateFormat.parse(PATIENT_BIRTHDATE);
+        long time = date.getTime();
+        Timestamp dob = new Timestamp(time);
+
+        Patient pat = new Patient();
+        pat.setBirthDate(dob);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        pat.setBirthDate(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(PATIENT_BIRTHDATE).getTime()));
```

File: src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,440 @@
...
+        DateFormat dateFormat2 = new SimpleDateFormat("dd/MM/yyyy");
+        Date date2 = dateFormat2.parse(SAMPLE_RECEIVED_TIMESTAMP);
+        long time2 = date2.getTime();
+        Timestamp doc = new Timestamp(time2);
+
+        java.sql.Date enteredDate = java.sql.Date.valueOf(SAMPLE_ENTERED_DATE);
+
+        Sample samp = new Sample();
+        samp.setEnteredDate(enteredDate);
+        samp.setReceivedTimestamp(doc);
```
Comment (mherman22) [inline src/test/java/org/openelisglobal/samplehuman/SampleHumanServiceTest.java]:
```suggestion
        samp.setReceivedTimestamp(new Timestamp(new SimpleDateFormat("dd/MM/yyyy").parse(SAMPLE_RECEIVED_TIMESTAMP).getTime()));
```

**Review by mherman22 — CHANGES_REQUESTED** [review]
@josephbate this looks good,

just a few minor suggestions below:

**Review by mherman22 — APPROVED** [review]
Thanks @josephbate , LGTM

cc: @mozzy11

**Comment by mozzy11** [comment]
Let me close this since it implemented here https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/1312

**Pull request closed without merging**